lib/internal/test_runner/reporter/spec.js

```diff
@@ -35,26 +34,24 @@ const symbols = {
   'arrow:right': '\u25B6 ',
   'hyphen:minus': '\uFE63 ',
 };
-class SpecReporter extends Transform {
-  #stack = [];
-  #reported = [];
-  #indentMemo = new SafeMap();
-  #failedTests = [];
 
-  constructor() {
-    super({ writableObjectMode: true });
-  }
+async function* specReporter(source) {
```

Comment:
If we want `new specReporter` to keep working (and I think we do, otherwise it's a breaking change), we cannot use a generator here, it has to be a `function specReporter(source)`, async functions and (async) generators are not "constructable".

Reply:
Just thinking out load, Is it possible to make `specReporter` function work consistently whether we use `new specReporter()` or just `specReporter`? I tried by it but it only works one way, can you suggest anything?

Reply:
That's what I meant, you need to use `function specReporter(source) { /* … */ }`, AFAIK it's the only thing is JS that is both callable and constructible.

Reply:
ok sorry, I got confused, so I did made it like `function specReporter(source) { /* … */ }` and it works with `new spec()` and also work as `spec()`, but does not work standalone like the `dot` and `tap`. viz:
```
run(...).compose(new spec()); // this works
run(...).compose(spec()); // this works
run(...).compose(spec); // this does not work
run(...).compose(tap); // <- tap and dot work like this
```
the third one above is used like the `dot` and `tap`. If `spec()` is also acceptable, let me know I'll push the change. Or let me know if my understanding is correct or not.

Reply:
Can you share what's the error you're getting? That would help understand what might be the problem.